eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1333,42 +1345,36 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
-            } else {
-              // this was a zero-conf splice and we already sent our splice_locked
-              None
-            }
-            maybeEmitEventsPostSplice(d.shortIds, d.commitments, commitments1)
+          // We check if this commitment was already locked before receiving the event (which happens when using 0-conf
+          // or for the initial funding transaction). If it was previously not locked, we must send splice_locked now.
+          val previouslyNotLocked = d.commitments.all.exists(c => c.fundingTxId == commitment.fundingTxId && c.localFundingStatus.isInstanceOf[LocalFundingStatus.NotLocked])
+          val spliceLocked_opt = if (previouslyNotLocked) Some(SpliceLocked(d.channelId, w.tx.txid)) else None
+          // If the channel is public and we've received the remote splice_locked, we send our announcement_signatures
+          // in order to generate the channel_announcement.
+          val remoteLocked = commitment.fundingTxIndex == 0 || d.commitments.all.exists(c => c.fundingTxId == commitment.fundingTxId && c.remoteFundingStatus == RemoteFundingStatus.Locked)
+          val localAnnSigs_opt = if (d.commitments.announceChannel && remoteLocked) commitment.signAnnouncement(nodeParams, d.commitments.params) else None
+          localAnnSigs_opt.foreach(annSigs => announcementSigsSent += annSigs.shortChannelId)
+          // If we've already received the remote announcement_signatures, we're now ready to process them.
+          localAnnSigs_opt.flatMap(ann => announcementSigsStash.get(ann.shortChannelId)).foreach(self ! _)
+          if (commitment.fundingTxIndex > 0) {
+            maybeEmitEventsPostSplice(d.aliases, d.commitments, commitments1)
             maybeUpdateMaxHtlcAmount(d.channelUpdate.htlcMaximumMsat, commitments1)
-            stay() using d.copy(commitments = commitments1) storing() sending toSend.toSeq
           }
+          stay() using d.copy(commitments = commitments1) storing() sending spliceLocked_opt.toSeq ++ localAnnSigs_opt.toSeq
         case Left(_) => stay()
       }
 
     case Event(msg: SpliceLocked, d: DATA_NORMAL) =>
       d.commitments.updateRemoteFundingStatus(msg.fundingTxId) match {
-        case Right((commitments1, _)) =>
-          maybeEmitEventsPostSplice(d.shortIds, d.commitments, commitments1)
+        case Right((commitments1, commitment)) =>
+          // If the commitment is confirmed, we were waiting to receive the remote splice_locked before sending our announcement_signatures.
+          val localAnnSigs_opt = if (d.commitments.announceChannel) commitment.signAnnouncement(nodeParams, commitments1.params) else None
+          localAnnSigs_opt.foreach(annSigs => announcementSigsSent += annSigs.shortChannelId)
+          // If we've already received the remote announcement_signatures, we're now ready to process them.
+          localAnnSigs_opt.flatMap(ann => announcementSigsStash.get(ann.shortChannelId)).foreach(self ! _)
```

Comment:
Just a refactoring, I find it easier to understand that way (the same could be done in the `WatchFundingConfirmedTriggered` handler):

```suggestion
          val localAnnSigs_opt = if (d.commitments.announceChannel) commitment.signAnnouncement(nodeParams, d.commitments.params) else None
          localAnnSigs_opt match {
            case Some(localAnnSigs) =>
              // The commitment was locked on our side and we were waiting to receive the remote splice_locked before sending our announcement_signatures.
              announcementSigsSent += localAnnSigs.shortChannelId
              // If we've already received the remote announcement_signatures, we're now ready to process them.
              announcementSigsStash.get(localAnnSigs.shortChannelId).foreach(self ! _)
            case None => // The channel is private or the commitment isn't locked on our side
          }
```